### ISSUE TITLE
Is the result of part segmentation using a vote strategy?

### ISSUE THREAD

**Issue by xindeng98** [guochengqian/PointNeXt#58]
Thanks for your codebase! The results reported in your paper do not contain relevant information. I would like to know the difference between your models using or not using vote strategy.

**Comment by zouyanmei** [comment]
Have you reproduced the current codebased on Scannet and shapenetpart? I have reproduced the code on scannet at 70.5 mIoU, which is much lower than the results mentioned by the authors.​

**Comment by xindeng98** [comment]
> Have you reproduced the current codebased on Scannet and shapenetpart? I have reproduced the code on scannet at 70.5 mIoU, which is much lower than the results mentioned by the authors.​

The authors do not report the results on scannet, but the results on shapenetpart are reproducible.

**Comment by xindeng98** [comment]
The training time of c160 in partseg is too long, I only test -s model, I hope the author open .log instead of only .ckpt.

**Comment by zouyanmei** [comment]
> > Have you reproduced the current codebased on Scannet and shapenetpart? I have reproduced the code on scannet at 70.5 mIoU, which is much lower than the results mentioned by the authors.​
> 
> The authors do not report the results on scannet, but the results on shapenetpart are reproducible.

Do you train from scratch or use pre-trained models? I have trained from scratch but only get 85% instance mIoU on ShapeNetPart.

**Comment by zouyanmei** [comment]
> > Have you reproduced the current codebased on Scannet and shapenetpart? I have reproduced the code on scannet at 70.5 mIoU, which is much lower than the results mentioned by the authors.​
> 
> The authors do not report the results on scannet, but the results on shapenetpart are reproducible.

The authors mention ScanNet's results of around 72% mIoU in other issues.

**Comment by xindeng98** [comment]
> > > Have you reproduced the current codebased on Scannet and shapenetpart? I have reproduced the code on scannet at 70.5 mIoU, which is much lower than the results mentioned by the authors.​
> > 
> > 
> > The authors do not report the results on scannet, but the results on shapenetpart are reproducible.
> 
> Do you train from scratch or use pre-trained models? I have trained from scratch but only get 85% instance mIoU on ShapeNetPart.

![image](https://user-images.githubusercontent.com/104637328/193732085-a29f35ab-348f-4269-b42f-c8cbc4759cb4.png)
the intermediate result trained from scratch, -c160 model, so I hope the authors open .log file.  I would like to get logs of baseline to refer to the quality of my code.

**Comment by zouyanmei** [comment]
> > > > Have you reproduced the current codebased on Scannet and shapenetpart? I have reproduced the code on scannet at 70.5 mIoU, which is much lower than the results mentioned by the authors.​
> > > 
> > > 
> > > The authors do not report the results on scannet, but the results on shapenetpart are reproducible.
> > 
> > 
> > Do you train from scratch or use pre-trained models? I have trained from scratch but only get 85% instance mIoU on ShapeNetPart.
> 
> ![image](https://user-images.githubusercontent.com/104637328/193732085-a29f35ab-348f-4269-b42f-c8cbc4759cb4.png) the intermediate result trained from scratch, -c160 model, so I hope the authors open .log file. I would like to get logs of baseline to refer to the quality of my code.

I got similar results like you. And according to my experiments about PointNeXt, the result is convergent now.

**Comment by guochengqian** [comment]
@xindeng98   @zouyanmei 
1. for any questions related to shapenetpart, please ask Yuchen. I will not take any question related this dataset anymore.
2. my suggestion is not to work on this dataset. All models achieve similar results. Many recent methods do not opensource code for shapenetpart anyway. 
3. I have no idea what happens to your training. it is wired to get a result so low. 
4. here is our training log: https://drive.google.com/file/d/1cLMKuidYGtGN2gVDHeU_ZQDK9YNlW5Sb/view?usp=sharing

**Comment by guochengqian** [comment]
@zouyanmei did you run test? note validation in training time is not the val miou. 
for scannet, we can easily reach over 71 val miou without any voting.

**Comment by zouyanmei** [comment]
> @zouyanmei did you run test? note validation in training time is not the val miou. for scannet, we can easily reach over 71 val miou without any voting.

Yes, and I got 70.5 val miou.

**Comment by zouyanmei** [comment]
Sorry, I only train for 1 time. And here is my training log: https://drive.google.com/file/d/19WPFqi63U_ZndFFfiKPvERe4aPEDzImi/view.

**Comment by guochengqian** [comment]
@zouyanmei  you only used 1 GPU. but the default cfg is for 8 GPUs. You have to manually tune it. A good start is to follow the linear lr rule.

**Comment by zouyanmei** [comment]
> @zouyanmei please do the following:
> 
> 1. confirm you used the latest code including latest openpoints
> 2. if yes to 1, then train for 3 times, and report the average value
> 3. if you are not able to achieve over 71, share your training command with me

And I also noticed that label smoothing does not used in ScanNet, which is proved efficient in S3DIS.

**Comment by guochengqian** [comment]
@zouyanmei a good catch. we found it does not matter use it or not. no big difference. and scannet is just a quick experiment we did in rebuttal period.

**Comment by zouyanmei** [comment]
> @zouyanmei you only used 1 GPU. but the default cfg is for 8 GPUs. You have to manually tune it. A good start is to follow the linear lr rule.
Thank you so much, I got it.

**Comment by xindeng98** [comment]
![image](https://user-images.githubusercontent.com/104637328/194259375-b14c2780-a6ee-4ba2-8acd-363fa80db21e.png)
![image](https://user-images.githubusercontent.com/104637328/194259657-3896cde0-733e-4efb-b0de-d179b627897b.png)
![image](https://user-images.githubusercontent.com/104637328/194260346-f0f743c8-aec4-4d05-b4cd-86c62fd7f780.png)
I found out that your configuration file has a problem, the configuration file says batch_size=8, REDEME says use single gpu, but in the .log file you use four gpus, so the actual batch_size should be 32. I wonder if this difference exists? @yuchenlichuck

**Comment by Roywangj** [comment]
Hi @guochengqian ,
There are two datasets 'shapenetcore_partanno_segmentation_benchmark_v0_normal' dataset to train shapenetpart' and hdf5_data. And in PointNext you use the former. Have you tried the later dataset and get a result, if yes, can you share the dataset code and corresponding configs. When I train model using the later one on single 3090, i can only get 84.8%mIOU， which is much worse than 86.5%.  Here is my dataset code, config, and log.
`@DATASETS.register_module()
class ShapeNetPart(Dataset):
    classes = ['airplane', 'bag', 'cap', 'car', 'chair',
               'earphone', 'guitar', 'knife', 'lamp', 'laptop',
               'motorbike', 'mug', 'pistol', 'rocket', 'skateboard', 'table']
    seg_num = [4, 2, 2, 4, 4, 3, 3, 2, 4, 2, 6, 2, 3, 3, 3, 3]


    cls_parts = {'earphone': [16, 17, 18], 'motorbike': [30, 31, 32, 33, 34, 35], 'rocket': [41, 42, 43],
                   'car': [8, 9, 10, 11], 'laptop': [28, 29], 'cap': [6, 7], 'skateboard': [44, 45, 46], 'mug': [36, 37],
                   'guitar': [19, 20, 21], 'bag': [4, 5], 'lamp': [24, 25, 26, 27], 'table': [47, 48, 49],
                   'airplane': [0, 1, 2, 3], 'pistol': [38, 39, 40], 'chair': [12, 13, 14, 15], 'knife': [22, 23]}
    cls2parts = []
    cls2partembed = torch.zeros(16, 50)
    for i, cls in enumerate(classes):
        idx = cls_parts[cls]
        cls2parts.append(idx)
        cls2partembed[i].scatter_(0, torch.LongTensor(idx), 1)
    part2cls = {}  # {0:Airplane, 1:Airplane, ...49:Table}
    for cat in cls_parts.keys():
        for label in cls_parts[cat]:
            part2cls[label] = cat

    def __init__(self,
                 data_root='data/shapenetpart',
                 num_points=2048,
                 split='train',
                 class_choice=None,
                 shape_classes=16, transform=None, **kwargs):
        self.data, self.label, self.seg = load_data_partseg(split, data_root)
        self.cat2id = {'airplane': 0, 'bag': 1, 'cap': 2, 'car': 3, 'chair': 4,
                       'earphone': 5, 'guitar': 6, 'knife': 7, 'lamp': 8, 'laptop': 9,
                       'motor': 10, 'mug': 11, 'pistol': 12, 'rocket': 13, 'skateboard': 14, 'table': 15}
        self.seg_num = [4, 2, 2, 4, 4, 3, 3, 2, 4, 2, 6, 2, 3, 3, 3, 3]
        self.index_start = [0, 4, 6, 8, 12, 16,
                            19, 22, 24, 28, 30, 36, 38, 41, 44, 47]
        self.num_points = num_points
        self.partition = split
        self.class_choice = class_choice
        self.transform = transform

        if self.class_choice != None:
            id_choice = self.cat2id[self.class_choice]
            indices = (self.label == id_choice).squeeze()
            self.data = self.data[indices]
            self.label = self.label[indices]
            self.seg = self.seg[indices]
            self.seg_num_all = self.seg_num[id_choice]
            self.seg_start_index = self.index_start[id_choice]
        else:
            self.seg_num_all = 50
            self.seg_start_index = 0
            self.eye = np.eye(shape_classes)

    def __getitem__(self, item):
        pointcloud = self.data[item][:self.num_points]
        label = self.label[item]
        seg = self.seg[item][:self.num_points]
        if self.partition == 'trainval':
            pointcloud = translate_pointcloud(pointcloud)
            indices = list(range(pointcloud.shape[0]))
            np.random.shuffle(indices)
            pointcloud = pointcloud[indices]
            seg = seg[indices]

        # this is model-wise one-hot enocoding for 16 categories of shapes
        feat = np.transpose(self.eye[label, ].repeat(pointcloud.shape[0], 0))
        data = {'pos': pointcloud,
                # 'x': feat,
                'cls': label,
                'y': seg}
        if self.transform is not None:
            data = self.transform(data)
        return data

    def __len__(self):
        return self.data.shape[0]
`

‘’‘
model:
  NAME: BasePartSeg
  encoder_args:
    NAME: PointNextEncoder
    blocks: [ 1, 1, 1, 1, 1 ] # 1, 1, 1, 2, 1 is better, but not the main focus of this paper
    strides: [ 1, 2, 2, 2, 2 ]
    width: 32
    in_channels: 4  # better than 4,6
    sa_layers: 3  # better than 2
    sa_use_res: True
    radius: 0.1
    radius_scaling: 2.5
    nsample: 32 # will not improve performance.
    expansion: 4
    aggr_args:
      feature_type: 'dp_fj'
    reduction: 'max'
    group_args:
      NAME: 'ballquery'
      normalize_dp: True
    conv_args:
      order: conv-norm-act
    act_args:
      act: 'relu' # leakrelu makes training unstable.
    norm_args:
      norm: 'bn'  # ln makes training unstable
  decoder_args:
    NAME: PointNextPartDecoder
    cls_map: curvenet
  cls_args:
    NAME: SegHead
    global_feat: max,avg  # apped global feature to each point feature
    num_classes: 50
    in_channels: null
    norm_args:
      norm: 'bn'


lr: 0.001
min_lr: null
optimizer:
  NAME: adamw
  weight_decay: 1.0e-4  # the best

criterion_args:
  NAME: Poly1FocalLoss


epochs: 300
sched: multistep
decay_epochs: [210, 270]
decay_rate: 0.1
warmup_epochs: 0

feature_keys: pos,heights

datatransforms:
  train: [PointsToTensor, PointCloudScaling,PointCloudCenterAndNormalize,PointCloudJitter]
  val: [PointsToTensor, PointCloudCenterAndNormalize]
  kwargs:
    jitter_sigma: 0.001
    jitter_clip: 0.005
    scale: [0.8, 1.2]
    gravity_dim: 1
    angle: [0, 1.0, 0]

dataset:
  common:
    NAME: ShapeNetPart
    data_root: data/ShapeNetPart/hdf5_data
    num_points: 2048
  train:
    split: trainval
  val:
    split: test
    presample: True
’‘’


![log](https://user-images.githubusercontent.com/48822865/219260603-eef9d0a5-45d4-4c9d-b75d-b62a9a8fbccd.png)
